imx/deposit.go

```diff
@@ -136,7 +135,7 @@ func (d *ETHDeposit) Deposit(ctx context.Context, c *Client, l1signer L1Signer,
 	if isRegistered {
 		return c.depositEth(ctx, l1signer, starkKey, big.NewInt(int64(signableDeposit.VaultId)), assetType, amount, overrides)
 	}
-	return c.registerAndDepositEth(ctx, l1signer, starkKeyHex, starkKey, big.NewInt(int64(signableDeposit.VaultId)), assetType, amount, overrides)
+	return c.registerAndDepositEth(ctx, l1signer, signableDeposit.StarkKey, starkKey, big.NewInt(int64(signableDeposit.VaultId)), assetType, amount, overrides)
```

Comment:
why `signableDeposit.StarkKey` and `starkKey` being passed in, what's the difference?

Reply:
that's probably why 
```
	// Passing starkKey string in hex format to register method because it may be padded and converting back from Int loses the padding
```

Reply:
Will add a todo and get back to this low priority refactor.

Reply:
Addressed